app/demo-loop/LDemoIO.cc

```diff
@@ -247,34 +232,8 @@ TransporterInput load_input(const LDemoArgs& args)
 
     // Create geometry/material coupling
     {
-        GeoMaterialParams::Input input;
-        input.geometry  = params.geometry;
-        input.materials = params.material;
-
-        input.volume_to_mat.resize(imported_data.volumes.size());
-        for (auto volume_idx :
-             range<VolumeId::size_type>(input.volume_to_mat.size()))
-        {
-            input.volume_to_mat[volume_idx]
-                = MaterialId{imported_data.volumes[volume_idx].material_id};
-        }
-        if (!volumes_are_consistent(*input.geometry, imported_data.volumes))
-        {
-            // Volume names do not match exactly between exported ROOT file and
-            // the geometry (possibly using a ROOT/GDML input with an ORANGE
-            // geometry): try to let the GeoMaterialParams remap them
-            CELER_LOG(warning) << "Volume/material mapping is inconsistent "
-                                  "between Geant4 data and geometry file: "
-                                  "attempting to remap";
-            input.volume_labels.resize(imported_data.volumes.size());
-            for (auto volume_idx : range(imported_data.volumes.size()))
-            {
-                input.volume_labels[volume_idx] = Label::from_geant(
-                    imported_data.volumes[volume_idx].name);
-            }
-        }
-        params.geomaterial
-            = std::make_shared<GeoMaterialParams>(std::move(input));
+        params.geomaterial = GeoMaterialParams::from_import(
```

Comment:
Much nicer! Remapping internally also makes more sense. I see you removed the warning, is there any chance that we might miss having the message in case the label remapping ends up being the problem?

Reply:
I don't think we'll miss the warning: in fact, knowing what I know now, it's only a "coincidence" when the names and IDs match up, because Geant4 and VecGeom build their models independently.

Reply:
Interesting. They do, but when I ran a few tests awhile ago comparing both, I've seen differences in IDs between programmatic and g4 / vecgeom. But loading a gdml file on g4 or vecgeom resulted in equivalent IDs...

Reply:
It generally seems to for the simple ones, but they definitely don't match up for CMS2018.